### PR TITLE
bb_quibbles

### PR DESCRIPTION
**Pull request by Boruch-Baum**
Simplifications:
+ remove need to perform nreverse operations
+ simplify 'setq x (cons (cons ... ) x' to use push
+ use replace-regexp-in-string
+ remove un-needed hash table

### CODE REVIEW

File: w3m-form.el [hunk 1]
```diff
@@ -439,7 +440,7 @@ fid=\\([^/]+\\)/type=\\([^/]+\\)/name=\\([^/]*\\)/id=\\(.*\\)$"
 		    (let ((hseq (nth 2 (w3m-action (point))))
 			  (value (w3m-form-get form id)))
 		      (when (> hseq 0)
-			(setq textareas (cons (cons hseq value) textareas)))
+			(push textareas (cons hseq value) textareas))
```
Comment (tsuchm) [inline w3m-form.el]: I think this change contains a mistake.  Do you mean (push (cons hseq value) textareas) ?

**Comment by tsuchm** [comment]
This pull request is too large.  At least, can you split the change related to octet.el and the changes related to others?

File: w3m-form.el (hunk 1 above)
Comment (Boruch-Baum) [reply w3m-form.el]: Yes! Thanks. Fixed.

**Comment by Boruch-Baum** [comment]
> This pull request is too large. At least, can you split the change related to octet.el and the changes related to others?

At this point I'm confused whether your request is still active because the github page has a few later statements that you "approved these changes". Does that mean that you cherry-picked and merged them, or does it mean something else and I should still try some way to divide them into two or more pull requests (probably by creating new branches)?

**Comment by Boruch-Baum** [comment]
@tsuchm: I see that you have approved the PR, but not merged it. In the meantime, I have prepared  a few other small optimizations for another push to this same branch bb_quibbles, but if this PR remains outstanding, it may seem to 'adopt' the newer commits to that branch.

This is a problem with my workflow that I need to solve, and would appreciate if you have advice. I suppose one would be just to multiply branches, but it would be nice to remain organized with all related commits on the same branch.

**Pull request merged**